UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs

```diff
@@ -710,6 +743,113 @@ private void DismemberBodyPartWithChance()
 			}
 		}
 
+		private void TakeBurnDamage(float burnDamage)
+		{
+			if(SelfArmor.Fire < burnDamage)
+			{
+				currentBurnDamage += burnDamage;
+				CheckBurnDamageLevels();
+			}
+		}
+
+		/// <summary>
+		/// Checks and sets what damage level this body part is on, once it becomes charred; the game displays it being charred.
+		/// </summary>
+		private void CheckBurnDamageLevels()
+		{
+			if (currentBurnDamage <= 0)
+			{
+				currentBurnDamageLevel = BurnDamageLevels.NONE;
+			}
+			if (currentBurnDamage >= 25)
+			{
+				currentBurnDamageLevel = BurnDamageLevels.MINOR;
+			}
+			if (currentBurnDamage >= 50)
+			{
+				currentBurnDamageLevel = BurnDamageLevels.MAJOR;
+			}
+			if (currentBurnDamage >= 75)
+			{
+				if(currentBurnDamageLevel != BurnDamageLevels.CHARRED) //So we can do this once.
+				{
+					var spritesList = Root.ImplantBaseSpritesDictionary.Values;
+					foreach (var sprites in spritesList)
+					{
+						foreach(var sprite in sprites)
+						{
+							sprite.baseSpriteHandler.SetColor(bodyPartColorWhenCharred);
+						}
+					}
+				}
+				currentBurnDamageLevel = BurnDamageLevels.CHARRED;
+				AshBodyPart();
+			}
+		}
+
+
+		/// <summary>
+		/// Turns this body part into ash while protecting items inside of that cannot be ashed.
+		/// </summary>
+		private void AshBodyPart()
+		{
+			if(currentBurnDamageLevel == BurnDamageLevels.CHARRED && currentBurnDamage > bodyPartAshesAboveThisDamage)
+			{
+				IEnumerable<ItemSlot> internalItemList = Storage.GetItemSlots();
+				IEnumerable<ItemSlot> PlayerItemList = healthMaster.PlayerScriptOwner.ItemStorage.GetItemSlots();
+				foreach(ItemSlot item in internalItemList)
+				{
+					if(item.ItemAttributes != null) //Incase this is an empty slot
+					{
+						if (item.ItemAttributes.CannotBeAshed)
+						{
+							Inventory.ServerDespawn(item);
+						}
+					}
+					var organ = item.ItemObject?.GetComponent<BodyPart>();
+					if (organ != null)
+					{
+						if (organ.DeathOnRemoval)
+						{
+							HealthMaster.Death();
+						}
+					}
+				}
+				if(PlayerItemList != null) //In case this is not a player
+				{
+					foreach (ItemSlot item in PlayerItemList)
+					{
+						if (item.ItemAttributes != null)
+						{
+							if (item.ItemAttributes.CannotBeAshed)
+							{
+								Inventory.ServerDrop(item);
+							}
```

Comment:
Could you also do and OR check for if its indestructable

Reply:
Done.